stetho/src/main/java/com/facebook/stetho/common/ListUtil.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.common;
+
+import java.util.AbstractList;
+import java.util.Collections;
+import java.util.List;
+
+public final class ListUtil {
+  private ListUtil() {
+  }
+
+  /**
+   * Compares two {@link List}s by comparing their contained items by object identity.
+   */
+  public static <T> boolean identityEquals(List<? extends T> list1, List<? extends T> list2) {
+    if (list1 == list2) {
+      return true;
+    }
+    int size = list1.size();
+    if (size != list2.size()) {
+      return false;
+    }
+    for (int i = 0; i < size; ++i) {
+      if (list1.get(i) != list2.get(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Copies the given {@link List} and returns the copy as an immutable {@link List}.
+   */
+  public static <T> List<T> copyToImmutableList(List<T> list) {
```

Comment:
Ack'd, though the problems with immutable list are more subtle than just encouraging copies.  One of the biggest issues is that they interoperate with List which is itself mutable.  Iterable, even, is mutable.

FWIW, I'd much rather have a strongly enforced annotation that prevents mutation through linting and runtime analysis but does nothing at runtime.  But Stetho isn't the place to solve this problem and that's why I prefer that Stetho just doesn't concern itself with immutability...